### ISSUE TITLE
Error: The block "xxx" can have a maximum of 3 keywords.

### ISSUE THREAD

**Issue by elliotcondon** [WordPress/gutenberg#11949]
**Describe the bug**
When registering a block type with more than 3 keywords, the following JS error is logged and the block type is not registered:
```
The block "xxx" can have a maximum of 3 keywords.
```

**To Reproduce**
1. Create a basic block type with 4 keywords
2. Check the console log

**Expected behavior**
The current behavior is too extreme.
The block type should be registered as normal, and WP should use the first 3 keywords ignoring any others if provided. There is also no need for a JS error, or even a warning.

**Comment by timelsass** [comment]
A silent fail would be nice for prod build, and when using dev build to provide the console warn or error.  Dev feedback should be more prominent

**Comment by elliotcondon** [comment]
Hi @timelsass. Can you clarify on what you mean by "silent fail"?
- Do you mean the block should fail to register? This seems overkill for such a minor setting.

**Comment by danielbachhuber** [comment]
> The block type should be registered as normal, and WP should use the first 3 keywords ignoring any others if provided.

I agree with this.

>  There is also no need for a JS error, or even a warning.

The warning could be helpful, but we could only show it in dev mode.

**Comment by jameelmoses** [comment]
PR submitted :crossed_fingers:

**Comment by timelsass** [comment]
@elliotcondon my bad I meant suppressing the error instead in production build - as it would still be a useful warning for people who are developing and registering blocks.

@jameelmoses should this now become console.warn since input is being accepted a modified instead of console.error?

**Comment by jameelmoses** [comment]
@timelsass yeah probably. just wasn’t sure if i was allowed to change linting rules, but seems fine if console.error is being disabled

**Comment by swissspidy** [comment]
Related discussion: #6633.

**Comment by aduth** [comment]
We should just abolish this 3-keyword limit. I haven't seen anyone provide any compelling reason for why it exist, and it's certainly caused more friction than value it's provided.

**Comment by aduth** [comment]
> PR submitted 🤞

Afore-mentioned pull request: #11953

**Comment by jameelmoses** [comment]
@aduth is this something you want me to tackle? or you guys want to handle internally? im not sure if there's a series of changes that require more thought and strategy, or if it's just in this section of code.

**Comment by aduth** [comment]
> @aduth is this something you want me to tackle? or you guys want to handle internally?

I'm not personally aware of any ongoing effort here, so feel free to take it on.

> im not sure if there's a series of changes that require more thought and strategy, or if it's just in this section of code.

I _think_ it's relatively well-contained here, but I'm happy to do some more digging if you ping me for a review on a related pull request.

**Comment by jameelmoses** [comment]
@aduth updated my PR to remove the restriction entirely

**Comment by gziolo** [comment]
Landed #13848 which removes this limit altogether.

**Comment by elliotcondon** [comment]
Good work @gziolo

**Comment by jameelmoses** [comment]
What am I, chopped liver @elliotcondon? ¯\_(ツ)_/¯ 

Jk ❤️

**Comment by elliotcondon** [comment]
@jameelmoses Haha, sorry mate. Just looked closer at the PR.
"All work was done by @jameelmoses."

Thank you sir.
